### PR TITLE
[REF] Remove unreachable lines

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
[REF] Remove unreachable lines

Before
----------------------------------------
Unreachable code present 

After
----------------------------------------
poof

Technical Details
----------------------------------------
The function recur ONLY calls single if $first is TRUE. Once we remove $first from the if clause
we can see it applies if recur is TRUE or if recur is FALSE (or any other variant given we are using
truthiness). So the if can go

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/Payment/PayPalIPN.php [hunk 1]
```diff
@@ -206,35 +206,27 @@ public function recur($input, $ids, $objects, $first) {
       return;
     }
```
Comment (eileenmcnaughton) [inline CRM/Core/Payment/PayPalIPN.php:207]: this return applies to if (!$first)

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
I took a quick look but it wasn't immediately obvious - can you explain a bit more why?

**Comment by eileenmcnaughton** [comment]
@mattwire didn't the technical details make sense? - 

The function recur ONLY calls single if $first is TRUE - so if recur is TRUE then $first is also TRUE - so $first can go & we are left with 
```
if ((!$recur) || ($recur)) {
```

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
I think it makes sense.

**Pull request merged**